### ISSUE TITLE
Video files to be stream instead of downloading and than playing

### ISSUE THREAD

**Issue by supportreq** [owncloud/android#592]
Hi,

Is it possible for the video files [mp4 h264] which are stored on owncloud, to be streamed directly, instead of them being downloaded and than played.

Using Android 4.4.4 Nexus 5 android app 1.5.8

____

PENDING TASKS:

- [X] [GIT] Create branch in Android repo
- [X] Use ExoPlayer library instead of VideoView because it was not possible to set headers for the authentication in < Lollipop devices.
- [X] Create custom http data source for setting authentication headers (ExoPlayer doesn't implement it).
- [X] Add full screen mode manually  (ExoPlayer doesn't implement it). 
* **Important**: This full screen mode is different to the one we previously had in the app. To go into full screen mode there is a new button in the playback controls and to exit the full screen mode, there is other button in the same place when full screen mode is enabled.
- [X] Handle full screen modes. Two full screen modes to implement depending on the Android versions:
* **< 4.4 KitKat:** Full screen with a limitation, the least user interaction will cause navigation controls to reappear immediately. Once the navigation controls have reappeared, they are not hidden again, even if the user clicks in the view.
* **\>= 4.4 KitKat:** App goes truly full screen using **immersive mode**, system bars are automatically hidden. User swipes to display the system bars.
- [X] Options in menu. Remove options not necessary to stream 
- [X] If the file is being downloaded you can not stream it, and if the user taps over the file a progress bar will be shown 
- [X] If new version is available it should works as always. Update the current file while play it. 
- [X] Detect SSL certificate to choose between download/stream. In this case, when the certificate is untrusted, an specific error message is shown to the user and after that, the streaming view is closed and the download starts automatically. 
- [X] Improve full screen button behaviour, avoiding show it till the player is ready to reproduce the video
- [X] Hide unnecessary options to stream in tablets [FIXED] @davigonz , @jesmrec 
- [X] Video playback state and autoplay are missed while rotating the screen in preview mode, not in full screen mode [FIXED] @davigonz @jesmrec 
- [X] Video playback state and autoplay are missed while changing from preview mode to full screen mode and vice versa [FIXED] @davigonz @jesmrec 
- [X] [QA] Design test plan @jesmrec
- [x] [QA] Validate test plan [WIP] @jesmrec
- [x] [GIT] Merge branch in Android repo

BUGS & IMPROVEMENTS

- [X] Crash when session expires https://github.com/owncloud/android/issues/592#issuecomment-282977504 [FIXED] @davigonz , @jesmrec 
- [X] Root folder when video can not be streamed https://github.com/owncloud/android/issues/592#issuecomment-282978338 [FIXED] @davigonz , @jesmrec 
- [X] Error message with no connection https://github.com/owncloud/android/issues/592#issuecomment-283001683  Remains in black screen ¿regression?. Check https://github.com/owncloud/android/issues/592#issuecomment-283408703 [FIXED] @davigonz , @jesmrec
- [X] Resume video after sharing https://github.com/owncloud/android/issues/592#issuecomment-283013464 @davigonz , @jesmrec  [FIXED]
- [X] Download and Streaming at the same time https://github.com/owncloud/android/issues/592#issuecomment-283018269  [FIXED] @davigonz , @jesmrec
- [X] Options missed https://github.com/owncloud/android/issues/592#issuecomment-283025413 [FIXED] @davigonz , @jesmrec 
- [X] Unsupported video files make the app crash https://github.com/owncloud/android/issues/592#issuecomment-283028375 [FIXED] @davigonz , @jesmrec 
- [X] Error message in slow connection https://github.com/owncloud/android/issues/592#issuecomment-283029422 [FIXED] @davigonz , @jesmrec 
- [X] Removed video makes crash https://github.com/owncloud/android/issues/592#issuecomment-283035219 [FIXED] @davigonz , @jesmrec 
- [X] Videos in flv shows a different error message https://github.com/owncloud/android/issues/592#issuecomment-283411879 [FIXED] @davigonz , @jesmrec 
- [X] Error when session expires in full screen https://github.com/owncloud/android/issues/592#issuecomment-283417772 [FIXED] @davigonz , @jesmrec 
- [X] Error message after remove a video in server and switch to full screen mode https://github.com/owncloud/android/issues/592#issuecomment-283581582 [FIXED] @davigonz , @jesmrec 
- [X] Crash in devices with 4.0.0 and 4.0.3 Ice Cream Sandwich versions while starting the video previewing or streaming https://github.com/owncloud/android/issues/592#issuecomment-286067156 [FIXED] @davigonz , @jesmrec 
- [X] https://github.com/owncloud/android/issues/859 [FIXED]
- [X] Different error messages for unsupported video files https://github.com/owncloud/android/issues/592#issuecomment-287054019 [WON'T FIX]
- [X] Expiration message is "not supported format" https://github.com/owncloud/android/issues/592#issuecomment-287055099 [WON`T FIX]
- [X] Should not-playable videos be downloaded always? https://github.com/owncloud/android/issues/592#issuecomment-287058618 [FIXED] @davigonz @jesmrec
- [X] Cancel button does not work https://github.com/owncloud/android/issues/592#issuecomment-287064011 [FIXED] @davigonz @jesmrec
- [X] Crash after downloading in 4.0.3 https://github.com/owncloud/android/issues/592#issuecomment-287303804 [FIXED] @davigonz @jesmrec
- [X] Crash in 4.0.3 while if orientation changes during download https://github.com/owncloud/android/issues/592#issuecomment-287331801 [FIXED] @davigonz @jesmrec
- [X] Video is streamed in av. offline https://github.com/owncloud/android/issues/592#issuecomment-287718525[FIXED] @davigonz @jesmrec
- [X] Error message in 302 redirections untrusted certificate https://github.com/owncloud/android/issues/592#issuecomment-287745048 [FIXED]
- [X] New version is downloaded and streamed https://github.com/owncloud/android/issues/592#issuecomment-287749600 [WONT_FIX]
- [X] Bad error message when forwarding video in expirated SAML session https://github.com/owncloud/android/issues/592#issuecomment-288007065 [READY_TO_TEST]
- [X] Download error in notification bar when session expires https://github.com/owncloud/android/issues/592#issuecomment-288007737

**Comment by xylo** [comment]
What about an option "stream with" below the option "open with" in the context menu?

**Comment by tobiasKaminsky** [comment]
@DeepDiver1975 Can you give us an hint, if and how streaming is possible via webdav?

**Comment by supportreq** [comment]
@AndyScherzinger can we have this?? Its the need of the hour as most cloud solutions have streaming of videos, mp3 instead of downloading...

**Comment by AndyScherzinger** [comment]
That is rather a question for @tobiasKaminsky, @DeepDiver1975 and other server side colleagues since I am rather a UI guy than a media streaming expert.

**Comment by tobiasKaminsky** [comment]
https://github.com/owncloud/core/issues/19958
So this is nearly a 10liner to create.
And then it will also work for audio :)

**Comment by AndyScherzinger** [comment]
Sounds awesome!!!

**Comment by supportreq** [comment]
This would be awesome.. Can we not implement streaming inside the android app itself cause we already have a video and audio player.. This will be an awesome and a long due feature..

**Comment by tobiasKaminsky** [comment]
No.
I tried it since 3 hours... The internal player is...awful.
I tried to play several downloaded files to just prove it is working before trying to stream them.
But none of the videos is being played.

So. I will just send the generated link via intent to all capable audio/video apps.

Supporting/enhancing the internal player is something, someone else can do :/

**Comment by supportreq** [comment]
I think the native player requires videos having native codec like h264, mp3 for it to work.. 

I guess what we can do is if it opens in the native player great else an option to open in another app.. what say?

**Comment by tobiasKaminsky** [comment]
Yes. I have convinced myself to give the internal player a second chance... ;)
The current state is:
- Passing the streaming url to the internal player
- If this does not work: 
 - Show warning that password can be exposed, then dialog yes/no, after that send url to vlc (later this will be a chooseable dialog)

The url has to include the password in cleartext, so a warning is necessary.

**Comment by tobiasKaminsky** [comment]
@all: Can someone please provide me a video sample that is working with the internal player? I am using a Nexus 4.

**Comment by tobiasKaminsky** [comment]
As this will be difficult to test as there are so many external players, different file names and different video formats, I will need some people to test, once the PR is ready.
I will then provide a apk with the PR, if this is ok for you? @davivel @rperezb

**Comment by supportreq** [comment]
@tobiasKaminsky I can test it for you.. 
Will share with you a video file.. please advice how do i send the file..

**Comment by tobiasKaminsky** [comment]
https://owncloud.tobiaskaminsky.de/index.php/s/kZ41pl8UPDieBQM
password: owncloud
There you can upload one or two videos.
Thank you!

**Comment by tobiasKaminsky** [comment]
The app is trying to stream the audio/video with the internal player, if this fails I show a dialog:
Sending the url to an external app can expose your password. Do you want to continue? -> Yes / No

Background: The url is in form of https://test:test@demo.owncloud.org/remote.php/webdav/Demo%20MP3%20-%20E.J.%20-%20Blick%20Zur%C3%BCck.mp3

@jancborchardt Is the wording ok for you?

**Comment by jancborchardt** [comment]
Wait – it can expose your password? How so?

The wording could be improved to:
> Sending the link to another app can expose your password. Do you want to continue?
> [Cancel] [Open in other app]

**Comment by tobiasKaminsky** [comment]
@jancborchardt Yes. The username and password must be part of the called url as of the "limitation" of webdav: it must have the credentials in the url to authenticate. 
The external player (e.g. vlc or mx player) only accepts a plain url and therefore we cannot send another authentication.

**Comment by jancborchardt** [comment]
@DeepDiver1975 @LukasReschke what do you think?

**Comment by xylo** [comment]
As long as you trust the external player the solution of tobiasKaminsky seems to be OK.

However, for future it would be nice if an alternative authentication method could be used, e.g.
* hash sum of user password and filename or
* one-time-password that is valid only for one day and the given file

**Comment by tobiasKaminsky** [comment]
I like the first idea of @xylo, but as far as I have read owncloud does not support http digest authentication: https://mail.kde.org/pipermail/owncloud/2013-June/009552.html

**Comment by supportreq** [comment]
@tobiasKaminsky do you require any other video file to test?
H264 codec with mp3 and atoom meta works fine..

can you share the PR so that we too can test it and help

**Comment by tobiasKaminsky** [comment]
Unfortunately your provided video cannot be played on my nexus 4...
The streaming PR is already in the beta version, which you can find here: https://github.com/owncloud/android/raw/beta/apks/owncloud-beta-2015-10-30.apk

**Comment by tobiasKaminsky** [comment]
My bad. Streaming is not included in beta so far.
Will do this in the nearer future.

**Comment by supportreq** [comment]
@tobiasKaminsky i checked the new beta which had streaming pr added.. it doesnt seem to work..

**Comment by tobiasKaminsky** [comment]
@supportreq Can you describe what you are doing when it is not working?
You can long-press on a video and select "stream externally".
But by default our internal player tries to stream the file, when you just click on it.
Then there is a (unfortunately) very long delay and after that you get the possibility to stream it with an external app.

**Comment by supportreq** [comment]
@tobiasKaminsky I am opening the video file and in the background it is just processing, the screen is stuck.. unable to press any button and the app crashes..

**Comment by supportreq** [comment]
![screenshot_2015-11-04-10-05-18](https://cloud.githubusercontent.com/assets/4680404/10930627/0732cd6c-82e6-11e5-96ae-2a3d03c25a6b.png)
![screenshot_2015-11-04-10-05-37](https://cloud.githubusercontent.com/assets/4680404/10930628/073386ee-82e6-11e5-8375-2ccbd2c66269.png)

**Comment by supportreq** [comment]
long press on the video gives the option for external viewing but it seems that it does not find the VLC app.. It gives me a list of browser apps, and when I select chrome it downloads the video file..

Show how the streaming function does not seem to work.. i m using one plus one..

**Comment by tobiasKaminsky** [comment]
Do you have the latest vlc version installed?
Can you try, for instance, MX player?
Indeed this is "just" a http(s) link to the file, therefore browsers may accept this link.

**Comment by supportreq** [comment]
I have the latest VLC from the Play Store.. Will check with MX player and revert..

**Comment by supportreq** [comment]
@tobiasKaminsky any update on this?? I am not sure if it is working even for mp3 files..

**Comment by supportreq** [comment]
Got this error : 
```
12-02 20:33:55.410 2803-5377/? E/ThumbnailsCacheManager: Generation of thumbnail for [id=6, name=VID-20151123-WA0000.mp4, mime=video/mp4, downloaded=true, local=/storage/emulated/0/owncloud/abc@192.168.1.113/Photos/VID-20151123-WA0000.mp4, remote=/Photos/VID-20151123-WA0000.mp4, parentId=3, favorite=false etag=40ec3b4a334ea126e676946510e2eec4] failed
                                                         java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
                                                             at com.owncloud.android.datamodel.ThumbnailsCacheManager.addVideoOverlay(ThumbnailsCacheManager.java:533)
                                                             at com.owncloud.android.datamodel.ThumbnailsCacheManager$ThumbnailGenerationTask.doInBackground(ThumbnailsCacheManager.java:243)
                                                             at com.owncloud.android.datamodel.ThumbnailsCacheManager$ThumbnailGenerationTask.doInBackground(ThumbnailsCacheManager.java:194)
                                                             at android.os.AsyncTask$2.call(AsyncTask.java:292)
                                                             at java.util.concurrent.FutureTask.run(FutureTask.java:237)
                                                             at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
                                                             at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
                                                             at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
                                                             at java.lang.Thread.run(Thread.java:818)
```

**Comment by tobiasKaminsky** [comment]
For me it is working with MX player, even with MP3.
But it needs some time...

Regarding the error, I have added some check and will upload a new beta soon.

**Comment by supportreq** [comment]
@tobiasKaminsky I verified with the latest beta : 20160427 on Android 6.0 oneplus CM13 it is still not working.. the video does not play

**Comment by supportreq** [comment]
`

**Comment by davivel** [comment]
Beta process is going to change. Current beta branch will not be maintained anymore. Video streaming is not available yet in the stable version, so this issue needs to be closed.

Thanks for reporting! :)

**Comment by davivel** [comment]
My mistake. This issue is feature request that we really want to have. Reopening.

**Comment by xylo** [comment]
@davivel 
I guess the removal of the "In progress" tag was a mistake, since there is a open pull request implementing this feature: https://github.com/owncloud/android/pull/1236
It's already waiting more than one year for approval.

**Comment by jesmrec** [comment]
### Bug

Steps

1. In a SAML server, wait until session expired
2. Tap on a not downloaded video 

Current behaviour

App crashes

Expected behaviour

Session has expired, and user is redirected to login view.

Tested with Huawei6P v7

CC @davigonz

**Comment by jesmrec** [comment]
### Improvement

Steps

1. Login in a non-trusted https server with a tablet
2. Set the tablet in landscape
3. Open a video that is in a folder (not in root)
4. Message is displayed

Current behaviour

Root folder is displayed

Expected behaviour

Folder that contains the video is displayed

Tested with Galaxy Tab S2 v5

CC @davigonz

**Comment by jesmrec** [comment]
Steps:

1. Stream a video
2. During the streaming, tap on "Share". Video is paused while share view is on the top.
3. Go back to player

Current behaviour

No way to play the video

Expected behaviour

Video should be played.

Tested with Huawei 6P v7, Samsung Galaxy Tab S2 v5

**Comment by jesmrec** [comment]
### Bug (or improvement?)

Steps

1. In a tablet, select a video and download
2. Set in landscape (download/details info is displayed in right fragment)
3. Set portrait

Current behaviour

Video is tried to be streamed at the same time is being downloaded. Not a bad behaviour but is spending more bandwidth as expected.

Expected behaviour

If it is being downloaded, should not be streamed.

Tested with Samsung Galaxy Tab S2 v5

**Comment by jesmrec** [comment]
### Bug

Options missed in menu:

Long press on a non-downloaded file in list view. The following options are missed:

- Set/Unset as Av. offline
- Send

Long press on a non-downloaded folder in list view. The following options are missed:

- Set/Unset as Av. offline

**Comment by jesmrec** [comment]
### Bug

Try to open some unsupported video formats (avi, wmv, ogv, flv)

Current behaviour

App crashes

Expected behaviour

File is downloaded

Tested with  Huawei 6P v7, Samsung Galaxy Tab S2 v5

**Comment by jesmrec** [comment]
**Improvement** related with this: https://github.com/owncloud/android/issues/592#issuecomment-283001683

This error message should be displayed only when the streaming can not start. 

If the streaming starts and the connection is slow, each time the streaming is stopped due to a lack of data, the message is displayed

**Comment by jesmrec** [comment]
### Bug

Steps

1. In server, delete a video file
2. In app, tap on the file (without refresh) that is not downloaded

Current Behaviour

App crash

Expected behaviour

Video can not be streamed

Tested with Huawei 6P v7, Samsung Galaxy Tab S2 v5

**Comment by davigonz** [comment]
@jesmrec 

> If the streaming starts and the connection is slow, each time the streaming is stopped due to a lack of data, the message is displayed

I'm not sure if this would be a suitable approach, it could be annoying to the user showing a message every time the streaming is working slow.

**Comment by jesmrec** [comment]
@davigonz yes, i meant this. If the connection is slow and the streaming stops and run, avoid showing the connection message

**Comment by jesmrec** [comment]
the message detected in this bug:

https://github.com/owncloud/android/issues/592#issuecomment-283001683

is also happening in this case:

1. Start to stream a video
2. Before the download finishes, switch the connection off
3. When the content can not be played, the correct error appears.
4. Switch to full screen -> appears the error with the video URL

It is a bit hidden ;)

**Comment by jesmrec** [comment]
Related with the unsupported formats.

Files in format .flv are not supported. If you try to stream a flv, you receive an `Unknown Error`instead a `Video cannot be reproduced`

**Comment by jesmrec** [comment]
Related with the session expired:

If the session expires (same as https://github.com/owncloud/android/issues/592#issuecomment-282977504) and a video is tried to be streamed in full screen, appears a long-non explicative error message that can frighten the user

**Comment by davigonz** [comment]
@jesmrec 

> yes, i meant this. If the connection is slow and the streaming stops and run, avoid showing the connection message

Sorry, I missunderstood you, I thinked that you was referring to showing the connection message as an improvement 😄

**Comment by jesmrec** [comment]
Related with file removed in server (https://github.com/owncloud/android/issues/592#issuecomment-283035219)

If the user changes to full screen one the video was deleted from server, he receives `Response Code 404`

**Comment by jesmrec** [comment]
@michaelstingl finally this feature will be included in 2.4.0. We prefer to release 2.3.0 asap due to https://github.com/owncloud/android/issues/1906, and work in some improvements to get a better UX in the streaming. 

CC @davigonz @davivel

**Comment by davigonz** [comment]
### Bug

Steps

1. In app, tap on a file not downloaded to stream it or already downloaded to preview it.
2. App tries to start the previewing.

Current Behaviour

App crash

Expected behaviour

Show an error message, and start the file download automatically.

Tested with Galaxy Nexus emulator with 4.0.0 Android version

**Comment by jesmrec** [comment]
For myself:

check https://github.com/owncloud/android/issues/859

**Comment by jesmrec** [comment]
Let's start a new QA review after the first bugfixing + code review.

**Comment by jesmrec** [comment]
I have noticed that in case of unsupported formats, the error messages are different. 

For example:

- flv shows: This video can not be reproduced
- wmv shows: This video format is not supported 

is this an expected behaviour? @davigonz @davivel

**Comment by jesmrec** [comment]
Following these steps

1. Start a SAML session
2. Wait until expires
3. Tap on a video (supported format)

Currently is shown: `This video format is not supported` and after tapping, user is redirected to login view.

The flow is perfect, but message is not appropiate for this case.

Tested with Nexus 10 v5

**Comment by jesmrec** [comment]
When the user tries to stream a video in an untrusted https server, a correct message is displayed and after that, the video starts to download.

When a user tries to stream an video in a unsupported format, a correct message is displayed (excepting https://github.com/owncloud/android/issues/592#issuecomment-287054019) and after that, nothing happens.

In the second case, the video should start to download like in the first one. The video format is not reproducible, but can be opened with an external app. From an UX POV is also better:

It would be a nice improvement: after downloading the not-played video, show the options to open it externally, like other kind of files in a dialog chart on the bottom of the screen

**Comment by jesmrec** [comment]
### Bug

Steps:

1. Login in a non-trusted https server
2. Tap on a video -> message indicating that video can not be played is shown. Video starts to download
3. Tap on (X) on the toolbar

Current behaviour

Video keeps on downloading

Expected behaviour

Downloading is cancelled.

Tested with Huawei 6P v7, Samsung Galaxy Note 4

Note: Changing orientation, the download is shown in details view. In this view, the (x) on toolbar works correctly.

**Comment by davigonz** [comment]
> I have noticed that in case of unsupported formats, the error messages are different.

> For example:

> flv shows: This video can not be reproduced
> wmv shows: This video format is not supported

> is this an expected behaviour?

Yes, this is an expected behaviour, the error message depends on the way the library handle the error. Flv files may trigger a different error than wmv files.

**Comment by davigonz** [comment]
> Following these steps

> Start a SAML session
 Wait until expires
 Tap on a video (supported format)
Currently is shown: This video format is not supported and after tapping, user is redirected to login view.

> The flow is perfect, but message is not appropiate for this case.

The point here is that SAML session is triggering the same error as the unsupported video format and we are not able to distinguish one from the other. The library decides the error message depending on the error is triggered, and in this case, both cases trigger the same error.

**Comment by davigonz** [comment]
> It would be a nice improvement: after downloading the not-played video, show the options to open it externally, like other kind of files in a dialog chart on the bottom of the screen

Sure, we have talked about this before and this is definitely a good improvement

**Comment by jesmrec** [comment]
### Bug

Steps

1. In Device with 4.0.3 (API 15), tap on a video
2. Error is displayed and video starts to download

Current behaviour

After downloading, the app crashes

Expected behaviour

The app not crashes.

Tested with Nexus 5 v4.0.3 (simulator)

**Comment by jesmrec** [comment]
### Bug

Steps

1. In Device with 4.0.3 (API 15), tap on a video
2. Error is displayed and video starts to download
3. While it is downloading, change orientation

Current behaviour

After downloading, the app crashes

Expected behaviour

The app not crashes.

Stack trace

03-17 12:32:36.983 23472-23472/com.owncloud.android E/AndroidRuntime: FATAL EXCEPTION: main
                                                                      java.lang.NullPointerException
                                                                          at com.owncloud.android.ui.preview.PreviewVideoFragment$1.onClick(PreviewVideoFragment.java:518)
                                                                          at android.support.v7.app.AlertController$ButtonHandler.handleMessage(AlertController.java:157)
                                                                          at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                          at android.os.Looper.loop(Looper.java:137)
                                                                          at android.app.ActivityThread.main(ActivityThread.java:4424)
                                                                          at java.lang.reflect.Method.invokeNative(Native Method)
                                                                          at java.lang.reflect.Method.invoke(Method.java:511)
                                                                          at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784)
                                                                          at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551)
                                                                          at dalvik.system.NativeStart.main(Native Method)

**Comment by jesmrec** [comment]
### Bug

Steps (only iPad in landscape)

1. Set a video as available offline
2. When it is downloading, cancel the download 
3. Tap on the video (streaming starts)
4. Pull down in files view

Current behaviour

video streams and downloads at the same time

Expected behaviour

video is donwloaded

CC @davigonz

**Comment by jesmrec** [comment]
### Bug

1. Open a server with 302 redirection and non-trusted certificate
2. Try to stream a video

Current behaviour

Error message: `Response code: 302`

Expected behaviour

A clearer error message

CC @davigonz

**Comment by jesmrec** [comment]
### Bug

1. In app, download a video
2. In server, upload a new version of the file 

Current behaviour

File is downloaded and streamed

Expected behaviour

File is downloaded

Tested with Galaxy Note4 v6 and Galaxy Tab v5

**Comment by davivel** [comment]
About error message in redirections, https://github.com/owncloud/android/issues/592#issuecomment-287745048

Straight redirections are followed, that was not the problem.

Crossed redirections are not allowed in streaming, that was resulting in the ugly error message. I updated it.

**Comment by davivel** [comment]
About the bug of streaming instead of downloading (https://github.com/owncloud/android/issues/592#issuecomment-287749600) 

the behaviour is correct. The video remote video is not being streamed, the local one is being played while the remote one is downloaded; this is consistent with the rest of the previews and the sync-on-click behaviour: the local copy of the file is shown while the remote one is downloaded. IMHO this is "won't fix".

**Comment by jesmrec** [comment]
### Bug
 
1. In a SAML server, start the streaming when the session is close to expire
2. When the session has expired (streaming keeps on), move forward the point in the progress bar to a place where the vide was not downloaded.

Current behaviour

Bad error message: EOF Exception

Expected behaviour

Appropiate error message

Tested with Galaxy Nexus v4.1.2

**Comment by jesmrec** [comment]
### Improvement

When a SAML session expires, the video starts to download and fails (notification bar). The error is not real, so the video does not have to download. The user sees an error that is the expected behaviour

**Comment by davivel** [comment]
@jesmrec , bug https://github.com/owncloud/android/issues/592#issuecomment-288007065 is ready to test.

About the improvement https://github.com/owncloud/android/issues/592#issuecomment-288007737 , to improve this behaviour we would need to make several important changes. This will delay us. I would discourage it.

**Comment by jesmrec** [comment]
All bugs fixed. Approved.

**Comment by davivel** [comment]
DONE; will be released in version 2.4.0, still no release date.

**Comment by davivel** [comment]
cc @michaelstingl
